lib/Config.ts

```diff
@@ -295,6 +297,36 @@ class Config {
   private getDefaultLogLevel = (): string => {
     return process.env.NODE_ENV === 'production' ? Level.Info : Level.Debug;
   }
+
+  private validateConfigProps(configProps: any) {
+    for (const prop in configProps) {
+      this.checkTypes(configProps, prop, (this as any)[prop]);
+    }
+  }
+
+  private checkTypes(configProps: any, prop: string, expectedVal: any) {
+    if (typeof configProps[prop] !== 'object') {
+      const expectedType = Object.prototype.toString.call(expectedVal);
+      const typeInConfigFile = Object.prototype.toString.call(configProps[prop]);
```

Comment:
`typeInConfigFile` is a little weird for a variable name. What about `actualType`? Would be a great contrast to `expectedType`?